lib/puppet/provider/package/yum.rb

```diff
@@ -25,6 +25,14 @@
 
   defaultfor :osfamily => :redhat
 
+  def execute(*args)
```

Comment:
I'm looking into this.  To provide this capability in a more general sense is going to require some more wide-spread modifications.  Specifically, I'll need to make modifications to the base provider class to allow providers to modify how the custom commands are executed.  I'm not sure enough about my ruby knowledge to be positive I'm doing this correctly, so I may ask for help after my next commit. :)